utils/yoke/src/yokeable.rs

```diff
@@ -259,18 +275,22 @@ where
         debug_assert!(mem::size_of::<Cow<'a, T>>() == mem::size_of::<Self>());
         let ptr: *const Self = (&from as *const Self::Output).cast();
         mem::forget(from);
-        core::ptr::read(ptr)
+        // Safety: `ptr` is certainly valid, aligned and points to a properly initialized value, as
+        // it comes from a mem::forget-ed
+        unsafe { core::ptr::read(ptr) }
```

Comment:
thought: we should actually move to ManuallyDrop instead of mem::forget, as is modern best practice

Reply:
I did that.